distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso

```diff
@@ -114,7 +114,15 @@ type Vector
     type Vector storage
 
     to_array self =
-        self.storage.to_array
+        arr = self.storage.to_array
+        case arr of
+            Array ->
+                arr
+            _     ->
+                len = self.storage.length
+                a = Array.new len
+                Array.copy arr 0 a 0 len
+                a
```

Comment:
Did you test this code? 

My experience tells me that polyglot arrays also used to match the `Array` case in pattern matching, so I'm not really sure if it will do what you want - it will most likely always enter the `Array` branch. Any array-object that is compatible with `Array.copy` will also match the `Array` branch, so the second branch is essentially dead code, unless I'm missing something.

See the following code:
```
from Standard.Base import all

polyglot java import java.util.List

main =
    listarr = List.of 1 2 3
    IO.println listarr
    m = Meta.meta listarr
    IO.println m
    IO.println m.get_language
    IO.println (Meta.get_simple_type_name listarr)
    IO.println (Meta.get_qualified_type_name listarr)
    case listarr of
        Array -> IO.println "Matches Array!"
        _ -> IO.println "nope"

```
it actually prints
```
[1, 2, 3]
(Polyglot [1, 2, 3])
Java
java.util.ImmutableCollections$ListN
null
Matches Array!
```

Reply:
Also, I think the `to_array` method should have some doc comment, if at least `## PRIVATE`.

Reply:
(Sorry for commenting after merge, but I did not notice these earlier)

Reply:
This was affecting my `Array.set_at` removal tests, so I fixed it in 29c4abc6ba9a29ddcac9bb5b686e53fcadfb2b69 (part of #3634).